x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -815,6 +821,143 @@ And the Investigation Guide tab should NOT be displayed
+
+**Automation**: 2 integration tests with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And <field_name> field is customized by the user (current version != base version)
+And <field_name> field is not updated by Elastic in this upgrade (target version == base version)
+Then for <field_name> field the diff algorithm should output the current version as the merged one without a conflict
+And <field_name> field should not be returned from the `upgrade/_review` API endpoint
+And <field_name> field should not be shown in the upgrade preview UI
+
+Examples:
+| field_name | base_version | current_version | target_version |
+| name       | "A"          | "B"             | "A"            |
+| risk_score | 1            | 2               | 1              |
+```
+
+#### **Scenario: Rule field has an update and doesn't have a custom value**
+
+**Automation**: 2 integration tests with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And <field_name> field is customized by the user (current version == base version)
+And <field_name> field is not updated by Elastic in this upgrade (target version != base version)
+Then for <field_name> field the diff algorithm should output the target version as the merged one without a conflict
+And <field_name> field should be returned from the `upgrade/_review` API endpoint
+And <field_name> field should be shown in the upgrade preview UI
+
+Examples:
+| field_name | base_version | current_version | target_version |
+| name       | "A"          | "A"             | "B"            |
+| risk_score | 1            | 1               | 2              |
+```
+
+#### **Scenario: Rule field has an update and a custom value that are the same**
+
+**Automation**: 2 integration tests with mock rules + a set of unit tests for the algorithm
+
+```Gherkin
+Given at least 1 prebuilt rule is installed in Kibana
+And for this rule there is a new version available
+And <field_name> field is customized by the user (current version != base version)
+And <field_name> field is updated by Elastic in this upgrade (target version != base version)
+And customized <field_name> field is the same as the Elastic update in this upgrade (current version == target version)
+Then for <field_name> field the diff algorithm should output the current version as the merged one without a conflict
+And <field_name> field should not be returned from the `upgrade/_review` API endpoint
+And <field_name> field should not be shown in the upgrade preview UI
+
+Examples:
+| field_name | base_version | current_version | target_version |
+| name       | "A"          | "B"             | "B"            |
+| risk_score | 1            | 2               | 2              |
```

Comment:
Nit: Just want to point out that this case is similar to the one described in https://github.com/elastic/kibana/issues/180154, and we might want to revisit it later. But for now let's keep it as is.